File: src/DesignPattern/Structural/Flyweight/Flyweight-cc/flyweight_Test.cc
#include <stdio.h>
#include "./flyweight.h"

int main(){
    auto f = new FlyweightFactory();
    auto item1 = f->GetFlyweight('a');
    auto item2 = f->GetFlyweight('a');
    auto item3 = f->GetFlyweight('b');
    printf("%d %p\n",item1->getValue(),item1);
    printf("%d %p\n",item2->getValue(),item2);
    printf("%d %p\n",item3->getValue(),item3);

    f->Clear();
    delete f;
}

File: src/DesignPattern/Structural/Flyweight/Flyweight-ts/flyweight.test.ts
import { FontFactory } from "./flyweight";

let f = new FontFactory();
let item1 = f.GetFont(10);
let item2 = f.GetFont(10);
let item3 = f.GetFont(11);

console.log(item1.getValue(),item2.getValue(),item3.getValue());
console.log(item1 == item2);